lib/app.dart

```diff
@@ -113,80 +121,84 @@ class _LanternAppState extends State<LanternApp> {
-                  initialLocale: currentLocale(lang),
-                  child: MaterialApp.router(
-                    locale: currentLocale(lang),
-                    debugShowCheckedModeBanner: false,
-                    theme: ThemeData(
-                      useMaterial3: false,
-                      fontFamily: _getLocaleBasedFont(currentLocal),
-                      brightness: Brightness.light,
-                      primarySwatch: Colors.grey,
-                      appBarTheme: const AppBarTheme(
-                        systemOverlayStyle: SystemUiOverlayStyle.dark,
+                  child: I18n(
+                    initialLocale: currentLocale(lang),
+                    child: MaterialApp.router(
+                      locale: currentLocale(lang),
+                      debugShowCheckedModeBanner: false,
+                      theme: ThemeData(
+                        useMaterial3: false,
+                        fontFamily: _getLocaleBasedFont(currentLocal),
+                        brightness: Brightness.light,
+                        primarySwatch: Colors.grey,
+                        appBarTheme: const AppBarTheme(
+                          systemOverlayStyle: SystemUiOverlayStyle.dark,
+                        ),
+                        colorScheme: ColorScheme.fromSwatch()
+                            .copyWith(secondary: Colors.black),
                       ),
-                      colorScheme: ColorScheme.fromSwatch()
-                          .copyWith(secondary: Colors.black),
-                    ),
-                    title: 'app_name'.i18n,
-                    localizationsDelegates: const [
-                      GlobalMaterialLocalizations.delegate,
-                      GlobalWidgetsLocalizations.delegate,
-                      GlobalCupertinoLocalizations.delegate,
-                    ],
-                    routerConfig: globalRouter.config(
-                      deepLinkBuilder: navigateToDeepLink,
+                      title: 'app_name'.i18n,
+                      localizationsDelegates: const [
+                        GlobalMaterialLocalizations.delegate,
+                        GlobalWidgetsLocalizations.delegate,
+                        GlobalCupertinoLocalizations.delegate,
+                      ],
+                      routerConfig: globalRouter.config(
+                        deepLinkBuilder: navigateToDeepLink,
+                      ),
+                      supportedLocales: const [
+                        Locale('ar', 'EG'),
+                        Locale('fr', 'FR'),
+                        Locale('en', 'US'),
+                        Locale('fa', 'IR'),
+                        Locale('th', 'TH'),
+                        Locale('ms', 'MY'),
+                        Locale('ru', 'RU'),
+                        Locale('ur', 'IN'),
+                        Locale('zh', 'CN'),
+                        Locale('zh', 'HK'),
+                        Locale('es', 'ES'),
```

Comment:
I think we want to keep `Locale('es', 'CU')`

Reply:
Added